### ISSUE TITLE
Process wait on timeout causes delayed health check failures on Windows

### ISSUE THREAD

**Issue by slackpad** [hashicorp/consul#3570]
The health check changes we made under https://github.com/hashicorp/consul/pull/3567 cause the health check status change to be delayed on Windows. We weren't able to terminate the process group, so on Windows we just wait for the process to finish after a timeout. This is good from the standpoint of not running another instance of the check concurrently, but it delays information about the check timing out.

We should probably add a call here to post the status while we are waiting for the process to exit (from check.go):

```
		if err := KillCommandSubtree(cmd); err != nil {
			c.Logger.Printf("[WARN] Timed out running check '%s': error killing process: %v", cmdDisplay, err)
		} else {
			c.Logger.Printf("[WARN] Timed out (%s) running check '%s'", timeout.String(), cmdDisplay)
		}
		c.Notify.UpdateCheck(c.CheckID, api.HealthCritical, err.Error()) // <- ADDED
```

For non-Windows we don't expect to hang out there for very long since we've killed the child processes, so we probably don't want the overhead of doing two `Notify` calls in succession, so a Windows-specific `if` around this might be warranted.

Another way to tackle this that might be more generic is to call the `Notify` like 1 or 2 seconds after we started waiting for `waitCh`. This will quickly fail the check if we get stuck there, but avoid the double write if we kill it or it exits close to the timeout.

**Comment by slackpad** [comment]
I'm thinking the second way is the cleaner way to fix this, rather than a `GOOS`-specific hack.

**Comment by slackpad** [comment]
Also thinking we should fix this before the 1.0 release, since this can cause significant delays vs. the previous behavior.

**Comment by slackpad** [comment]
We ended up just changing the timeout path to report the error immediately after the timeout, along with the output, which is simpler and works on all platforms.
